### PR TITLE
add default value for command & args in README document

### PR DESCRIPTION
**Pull request by NagleZhang**
add default value for command & args, command is `sleep`, and args default value is `99999999`.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -218,8 +218,8 @@ The `containerTemplate` is a template of container that will be added to the pod
 * **envVars** Environment variables that are applied to the container **(supplementing and overriding env vars that are set on pod level)**.
     * **envVar** An environment variable whose value is defined inline.
     * **secretEnvVar** An environment variable whose value is derived from a Kubernetes secret.
-* **command** The command the container will execute.
-* **args** The arguments passed to the command.
+* **command** The command the container will execute, default vaule is `sleep`.
```
Comment (Vlatombe) [inline README.md:221]:
```suggestion
* **command** The command the container will execute, default value is `sleep`.
```

**Review by Vlatombe — APPROVED** [review]

**Comment by jglick** [comment]
I do not think this is really true. These are the _suggested_ values. We decided not to actually set these as defaults: https://github.com/jenkinsci/kubernetes-plugin/pull/915#issuecomment-742626582

**Comment by NagleZhang** [comment]
> I do not think this is really true. These are the _suggested_ values. We decided not to actually set these as defaults: [#915 (comment)](https://github.com/jenkinsci/kubernetes-plugin/pull/915#issuecomment-742626582)

But what we are facing is this is not the _suggested_ value. 
I've describe this issue under this PR: 
  https://github.com/jenkinsci/kubernetes-plugin/pull/992#issuecomment-842131356

**Comment by jglick** [comment]
@NagleZhang all that #915 changed was suggested GUI defaults as shown in the **Pipeline Syntax** and global pod template config pages, as well as samples & docs. You need to explicitly select a `command` and `args` in YAML or Groovy if you do not wish to use the default container entrypoint. Therefore the text in this PR is incorrect, or at least misleading.

> what we are facing is this is not the _suggested_ value

How so? From https://github.com/jenkinsci/kubernetes-plugin/pull/992#issuecomment-842131356

> I changed the arguments without change command

which just sounds like a user error: you can set both of these, or neither, but it would rarely make any sense to set only one.

**Comment by NagleZhang** [comment]
> which just sounds like a user error: you can set both of these, or neither, but it would rarely make any sense to set only one.

Okay , this make sense. 
further more question is, 
what if we just change command without arguments ? ( I think this a normal usage for docker)

**Comment by jglick** [comment]
> if we just change command without arguments

It _could_ be legitimate on occasion—just depends on the image. Not likely I think.

There are several situations here:

* The default image for the `jnlp` container is already set up with an appropriate entrypoint, running agent stuff. There is no need to change command or arguments.
* If you use a nondefault image for `jnlp` you may or may not need to set command and arguments. For example, there is a single-container pod idiom by which you take a non-Jenkins-specific image with some tools, then add a JRE, plus `agent.jar` (and optionally also a stock agent launcher shell script) copied from the stock image. If the entrypoint is overridden to be Jenkins-specific in a derived image (`FROM generic-image`), then you would not specify command/args in the pod template; but if you use a truly stock image containing a JRE without derivation, and for example mount the agent JAR and launcher script from a `ConfigMap`, you would need to point to these mounts in the command/args of the pod template.
* Many images have no special entrypoint (just `bash` or the like), or run some sort of CLI command that would exit quickly. For these, you want to specify `sleep 99999999`: the container should just wait for the Pipeline to use `container('…') {sh '…'}`, which basically `kubectl exec`s into it. (Yes this is not how containers are supposed to be used, but it is how we can make them work with Jenkins agents.)
* Some images have an entrypoint which starts some sort of service. These might make sense if for example tests need to connect to a sample database or whatever, and you might never use the `container` step with them. For such cases there is no need to change command or arguments.

**Comment by NagleZhang** [comment]
> The default image for the jnlp container is already set up with an appropriate entrypoint, running agent stuff. There is no need to change command or arguments.

Yes, what we are using is https://github.com/jenkinsci/helm-charts. 
from the value they provide from this [line](https://github.com/jenkinsci/helm-charts/blob/e022f70dea4c5272c927f825af98e2d2c1482896/charts/jenkins/values.yaml#L631-L632)

the provide default command is `null` , but arguments do have like `  args: "${computer.jnlpmac} ${computer.name}"` 

and this values is used by [_helpers.tpl](https://github.com/jenkinsci/helm-charts/blob/e022f70dea4c5272c927f825af98e2d2c1482896/charts/jenkins/templates/_helpers.tpl#L208-L209) 

assume this is the issue from the our helm-chats ?

and , do you suggest we need more specific description for this configuration ? 
like we change line 
> default value is `sleep`
to 
> suggested value is `sleep`
?

**Comment by jglick** [comment]
I am not very familiar with that chart but it looks to apply only to the `jnlp` container, which would not normally need to have its `command` or `args` overridden—the image’s default entrypoint should be left alone.

**Comment by NagleZhang** [comment]
> I am not very familiar with that chart but it looks to apply only to the jnlp container, which would not normally need to have its command or args overridden—the image’s default entrypoint should be left alone. 

Yeah, That's what I'm think about as well. 

Thanks for your clarify.

**Pull request merged**